### DISCUSSION TITLE
Authentication steps not included for OAuth authentication

### DISCUSSION THREAD

**Discussion by Gvickers6** [grafana/grafana-infinity-datasource#936] [Q&A]
Hello all,

I'm getting a 401 unauthorized error message. Details are "Authentication steps not included for OAuth authentication"

Using OAuth as well.

I've configured OAuth and validated my permissions. I even went as far as pulling the same exact config into Postman. Generating the token, using the token, and the same POST works just fine in postman.

Auth is set to OAuth
Grant Type is Client credentials (same as postman)
Auth style is auto
The secrets and other settings are the same between postman and infinity.

Yet I can generate a token and query with postman. I get the message saying auth steps aren't included in Infinity. why? I'm stuck and have exausted going through every setting I see to try and figure this out. It seems Infinity has the settings for OAuth but is actively refusing to generate and use a token prior to querying anything in explore. I haven't even gotten to the point of generating a dashboard because I can't get the data source to give me anything.

**Comment by yesoreyeram** [comment]
@Gvickers6 - Sorry for the delay getting back. Can you explain which API you are using and what type of oauth2?  Currently infinity only supports oauth2 client_credentials/jwt authentication.
